### PR TITLE
Feat/signin

### PR DESCRIPTION
**Pull request by rat6666**

### CODE REVIEW

File: frontend/src/app/libs/container/store/session/session.reducer.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+export const sessionFeatureKey: 'session' = 'session';
+
+const initialState: ISessionState = {
+	user: null,
+	isLogged: false,
+};
+
+export const reducer: ActionReducer<ISessionState | undefined, Action> = (
```
Comment (clain-atomi) [inline frontend/src/app/libs/container/store/session/session.reducer.ts:16]: Please add separate type for Session actions, refer to container/store design

File: frontend/src/app/libs/sign-in/sign-in.component.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<form [formGroup]="signIn">
+
+  <label>
+    Email
+    <input type="email" formControlName="email" required autofocus>
```
Comment (clain-atomi) [inline frontend/src/app/libs/sign-in/sign-in.component.html:5]: please use angular material components, add necessary to common-components lib and go on :)

File: frontend/src/app/libs/sign-in/sign-in.component.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import { IUser } from '../container/models/user';
+import { SessionFacadeService } from '../container/store/session/session.facade';
+
+@Component({
+	selector: 'app-sign-in',
+	templateUrl: './sign-in.component.html',
+	styleUrls: ['./sign-in.component.scss'],
+})
+export class SignInComponent {
+	public user: IUser;
```
Comment (clain-atomi) [inline frontend/src/app/libs/sign-in/sign-in.component.ts:12]: Naming interfaces with starting 'I' is .net and java practice, please rename to `User` interface.

File: frontend/src/app/libs/sign-in/sign-in.component.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+	public signIn: FormGroup = new FormGroup({
```
Comment (clain-atomi) [inline frontend/src/app/libs/sign-in/sign-in.component.ts:14]: `signInForm` names need to be meaningfull

File: frontend/src/app/libs/sign-in/sign-in.component.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+		email: new FormControl('', [Validators.required, Validators.email]),
+		password: new FormControl('', [Validators.required]),
+	});
+
+	constructor(private sessionFacadeService: SessionFacadeService) {}
+
+	public submit(): void {
+		this.user = this.signIn.value;
```
Comment (clain-atomi) [inline frontend/src/app/libs/sign-in/sign-in.component.ts:22]: Is `signIn` value is really user? Seems like you just store username and password here, it;s not user, so create more meaningfull name like something with `interface UserCredentials { login: string, password: string }` and current this.user will become `this.credentials` or `this.userCredentials`

File: package-lock.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (clain-atomi) [inline package-lock.json:1]: do we really need this file here?

File: frontend/src/app/libs/container/store/session/session.facade.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+import { Signin } from './session.actions';
+
+@Injectable({
+	providedIn: 'root',
+})
+export class SessionFacadeService {
+	constructor(private store: Store<ISessionState>) {}
+
+	public signIn(user: IUser): void {
+		this.store.dispatch(new Signin(user));
```
Comment (clain-atomi) [inline frontend/src/app/libs/container/store/session/session.facade.ts:14]: typo, `SignIn(user)`

File: frontend/src/app/libs/container/store/session/session.reducer.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
+import { Action, ActionReducer } from '@ngrx/store';
+import { IUser } from '../../models/user';
+import { ActionType, Signin } from './session.actions';
+import { SessionActionTypes, SigninSucces } from './session.actions';
+
+export interface ISessionState {
```
Comment (clain-atomi) [inline frontend/src/app/libs/container/store/session/session.reducer.ts:6]: `SessionState`

File: frontend/src/app/libs/container/store/session/session.reducer.ts [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+	state: ISessionState = initialState,
+	action: SessionActionTypes
+) => {
+	switch (action.type) {
+		case ActionType.Signin:
+			return {
+				...state,
+				user: (action as Signin).payload,
+			};
+		case ActionType.SigninSucces:
```
Comment (clain-atomi) [inline frontend/src/app/libs/container/store/session/session.reducer.ts:26]: `SignInSuccess`

File: frontend/src/app/libs/container/store/store.module.ts [hunk 10]
```diff
@@ -1,15 +1,10 @@
...
+import { SessionEffects } from './session/session.effects';
 
 @NgModule({
-	imports: [
-		EffectsModule.forFeature([SampleEffects]),
-		StoreModule.forFeature(sampleFeatureKey, {
-			sample: reducer
-		})
-	]
+	imports: [EffectsModule.forFeature([SessionEffects]), StoreModule.forFeature(sessionFeatureKey, sessionStateReducer)],
```
Comment (clain-atomi) [inline frontend/src/app/libs/container/store/store.module.ts:8]: each store must be separate module, which inits in parent module so better revert this change and move session store init into sign-in.module.ts

File: frontend/src/app/libs/sign-in/mock-signin.service.ts [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+import { delay } from 'rxjs/operators';
+import { IUser, IRegisteredUser } from '../container/store';
+
+const delay100: number = 100;
+
+@Injectable({
+	providedIn: 'root',
+})
+export class MockSigninService {
+	private user: IRegisteredUser = {
```
Comment (clain-atomi) [inline frontend/src/app/libs/sign-in/mock-signin.service.ts:12]: Why do we need `IRegisteredUser` interface? Seems like it's just user and there are no need to use word `registered`.

File: frontend/src/app/app.component.html [hunk 12]
```diff
@@ -1,533 +1,4 @@
-<!-- * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * -->
```
Comment (clain-atomi) [inline frontend/src/app/app.component.html:1]: seems like your functionality doesn't affect this code, so why you deleted this component content?

File: frontend/angular.json [hunk 13]
```diff
@@ -1,143 +1,146 @@
...
+      "sourceRoot": "src",
+      "prefix": "app",
+      "architect": {
+        "build": {
+          "builder": "@angular-builders/custom-webpack:browser",
+          "options": {
+            "customWebpackConfig": {
+              "path": "./webpack.config.js"
             },
-            "root": "",
```
Comment (clain-atomi) [inline frontend/angular.json:13]: why there are so many changes in angular.json?

File: frontend/.prettierrc [hunk 14]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (clain-atomi) [inline frontend/.prettierrc:1]: remove your local prettier config from file.

Comment (rat6666) [reply frontend/.prettierrc]: done

File: frontend/src/app/app.component.html (hunk 12 above)
Comment (rat6666) [reply frontend/src/app/app.component.html]: done

File: frontend/angular.json (hunk 13 above)
Comment (rat6666) [reply frontend/angular.json:13]: it seems because I made a clone before the last two commits
